packages/material-ui/src/styles/colorManipulator.js

```diff
@@ -32,12 +32,9 @@ function clamp(value, min = 0, max = 1) {
  * @returns {string} A CSS rgb color string
  */
 export function hexToRgb(color) {
-  if (process.env.NODE_ENV !== 'production') {
-    if (!color) {
-      throw new Error(
-        `Material-UI: hexToRgb(color) is called with an invalid argument. (${color})`,
-      );
-    }
+  if (!color) {
+    warning(false, `Material-UI: missing color argument in hexToRgb(${color}).`);
+    return color;
```

Comment:
At this point it's not returning a string (unless it's an empty string) which is guaranteed by the documentation. I don't understand the use case for swallowing a TypeError here?

Reply:
I'm removing all the warnings. The discussion doesn't worth it.

Reply:
If people want meaningful errors, they can use TypeScript.